compose/cli/main.py

```diff
@@ -453,6 +437,10 @@ def exec_command(self, options):
             "stdin": tty,
         }
 
+        # setting environment for exec is not supported in API < 1.25'
+        if docker.utils.version_gte(self.project.client.api_version, '1.25'):
```

Comment:
We should call out that limitation in the docstring above, and raise a `UserError` if the API version is too low.